### DISCUSSION TITLE
v0.43.0

### DISCUSSION THREAD

**Discussion by tkoyama010** [pyvista/pyvista#5300] [Announcements]
## Highlights

* Allow some customization to trame toolbar by @FelipeCybis in https://github.com/pyvista/pyvista/pull/4855

https://github.com/pyvista/pyvista/assets/41338087/d6ba752a-b158-4441-83b0-3e638529c6ef

* Add 3D SolidSphere and SolidSphereGeneric as geometric objects by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/4965

<img src="https://github.com/pyvista/pyvista/assets/39341281/189f7e1b-ed31-431f-aaba-c6680b489a2d" width="480">

* Add radial gradient background mode by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4974

<img src="https://github.com/pyvista/pyvista/assets/7513610/f314ca7d-df5f-4bdd-aa45-3c66e3c1b925" width="480">

* Support user logos on the viewport by @banesullivan in https://github.com/pyvista/pyvista/pull/5103

<img width="480" alt="Screenshot 2023-10-18 at 8 24 40 PM" src="https://github.com/pyvista/pyvista/assets/22067021/47726ba7-bd2e-4bd1-9745-5f64b1c95c5c">

* Add more typehints by @akaszynski in https://github.com/pyvista/pyvista/pull/5171

* Add conversion of 3D label maps to labeled meshes with SurfaceNets. by @jmargeta in https://github.com/pyvista/pyvista/pull/5176

<img width="480" alt="Screenshot 2023-10-18 at 8 24 40 PM" src="https://github.com/pyvista/pyvista/assets/1239359/aa80a677-8c73-45bf-9475-7e331bce35e8">

### New Features
* Add class to HTML widget by @larsoner in https://github.com/pyvista/pyvista/pull/4847
* Allow some customization to trame toolbar by @FelipeCybis in https://github.com/pyvista/pyvista/pull/4855
* Update `connectivity` filter with new extraction modes and scalars by @user27182 in https://github.com/pyvista/pyvista/pull/4824
* Add `ConeSource` class by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4860
* Add `add_timer_event` helping make animation by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4839
* Move user matrix from `Actor` to `Prop3D` by @FelipeCybis in https://github.com/pyvista/pyvista/pull/4931
* Add `CylinderSource` class by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4890
* Add `MultipleLinesSource` class by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4954
* Add radial gradient background mode by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4974
* Add function `spherical_to_cartesian` to `features` by @denizkirbiyik in https://github.com/pyvista/pyvista/pull/5003
* Add `justification` property to `TextProperty` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5055
* Add 3D SolidSphere and SolidSphereGeneric as geometric objects by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/4965
* Add support for custom cyclers to color MultiBlocks by @abhiramtilakiiit in https://github.com/pyvista/pyvista/pull/5099
* Add `texture_coordinates` to `surface_from_para` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5098
* Add theme option to allow plotting empty meshes by @banesullivan in https://github.com/pyvista/pyvista/pull/5122
* Support user logos on the viewport by @banesullivan in https://github.com/pyvista/pyvista/pull/5103
* Add off screen devcontainer option by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5128
* Add option to get `figure_path` from environment variables by @AlejandroFernandezLuces in https://github.com/pyvista/pyvista/pull/5194
* Add option to always get screenshots from plotter shows by @AlejandroFernandezLuces in https://github.com/pyvista/pyvista/pull/5197
* Add conversion of 3D label maps to labeled meshes with SurfaceNets. by @jmargeta in https://github.com/pyvista/pyvista/pull/5176
* Add more typehints by @akaszynski in https://github.com/pyvista/pyvista/pull/5171
* Add `input_validation` package by @user27182 in https://github.com/pyvista/pyvista/pull/5100
* Add Text3DSource class by @user27182 in https://github.com/pyvista/pyvista/pull/5218
* Add `edge_opacity` property to `Property` class by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5192
* Add embeddable jupyter widget by @banesullivan in https://github.com/pyvista/pyvista/pull/5168
* Add `pack_labels` and `sort_labels` filters by @user27182 in https://github.com/pyvista/pyvista/pull/4802
* Add ability to control vertices for `PolyData` creation; allow all cell types to be used independently by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5245
* Add cast_to_polydata for Cell by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5242
* Improved Voxelization - Volume and Discretised Voxels by @bsburnham in https://github.com/pyvista/pyvista/pull/5095
* Add snap to closest point option for sample by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5271
### Bug fixes or behavior changes
* Deploy dev docs on non-tagged push by @akaszynski in https://github.com/pyvista/pyvista/pull/4841
* Fix Trame Viewers in Jupyter by @annehaley in https://github.com/pyvista/pyvista/pull/4844
* Fix typed marker file for mypy by @eggplants in https://github.com/pyvista/pyvista/pull/4863
* Fix Parametric objects direction error by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4853
* Make composite mapper dataset optional by @akaszynski in https://github.com/pyvista/pyvista/pull/4879
* Fix `geometric_objects.translate()` bug for collinear axes case by @user27182 in https://github.com/pyvista/pyvista/pull/4884
* Perform Render after Zoom event in default interactor key events by @ChristosT in https://github.com/pyvista/pyvista/pull/4958
* Fix `CylinderSource` orientation by @Arnav-2004 in https://github.com/pyvista/pyvista/pull/5004
* Give scraper a stable `__repr__` by @larsoner in https://github.com/pyvista/pyvista/pull/5013
* Fix module `vtkImageDilateErode3D` not found error by @Arnav-2004 in https://github.com/pyvista/pyvista/pull/5020
* Fix bug with actor removal rendering by @larsoner in https://github.com/pyvista/pyvista/pull/5046
* Remove trame from `all` in pyproject.toml, as it is no longer a variable by @jorgensd in https://github.com/pyvista/pyvista/pull/5072
* Captialize first word in sentence in `text.py` by @roceb in https://github.com/pyvista/pyvista/pull/5078
* Fix `Scraper` object `repr` and add a test of it by @adeak in https://github.com/pyvista/pyvista/pull/5022
* Fix issue with +/-Y face floors by @banesullivan in https://github.com/pyvista/pyvista/pull/5140
* Ensure Theme subclasses cannot receive arbitrary attributes after instantiation by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5112
* Remove set_plot_theme from examples by @banesullivan in https://github.com/pyvista/pyvista/pull/5144
* Fix rotation bug with 90 degrees on `rotate_vector` by @kmarchais in https://github.com/pyvista/pyvista/pull/5181
* Remove deprecated multi_samples keyword arg by @akaszynski in https://github.com/pyvista/pyvista/pull/5185
* Support button release events on interactor by @banesullivan in https://github.com/pyvista/pyvista/pull/5160
* Suppress VTK debug messages in contour_labeled and fix extract_all_edges logging level restoration by @jmargeta in https://github.com/pyvista/pyvista/pull/5228
* Prevent unnecessary copy in DataSet.rename_array by @darikg in https://github.com/pyvista/pyvista/pull/5246
* Fix test code to bump trame-vtk from 2.5.9 to 2.6.2 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5262
### Documentation
* Move version change step before doing test in release by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4834
* Fix pyvista.DataSetFilters.extract_points documentation does not represent the features of the method by @planmi1287 in https://github.com/pyvista/pyvista/pull/4856
* Add step to check tag before git push tag by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4858
* Update and fix `connectivity` docs by @user27182 in https://github.com/pyvista/pyvista/pull/4872
* Fix typos in docstring by @yr1404 in https://github.com/pyvista/pyvista/pull/4926
* Remove `ref` after references in documentation by @AdityaRanjanJha in https://github.com/pyvista/pyvista/pull/4946
* Move Slack to GitHub Discussions to announce by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4887
* Add Plotter.center and Plotter.image examples by @Arnav-2004 in https://github.com/pyvista/pyvista/pull/4993
* Fix errors of `codespell` v2.2.6 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5010
* Fix the name of `cartesian_to_spherical` 's returns by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5014
* Pin python in `intersphinx_mapping` to Python3.11 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5018
* Set `Google.Spacing` NO in `.vale.ini` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5027
* Define alias `pv` when importing `pyvista` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5034
* Add examples for widgets methods by @abhiramtilakiiit in https://github.com/pyvista/pyvista/pull/5036
* Add a Professional Support section by @jourdain in https://github.com/pyvista/pyvista/pull/5045
* Fix ParametricRandomHills Deprecation examples by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5050
* Move Professional Support section to Previous of Index Section by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5057
* Use title-style capitalization to fix vale warning by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5056
* Add `Text` and `TextProperty` to API documentation by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5058
* Fix the docstring link error which is using an alias by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5040
* Use `itertools.product` to avoid nested loops in examples by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5077
* Rename `master_doc` to `root_doc` in Sphinx build configuration file by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5107
* Fix the error of docstring example code in `open_movie` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5115
* Fix the link of docstring in `text.py` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5073
* Update AUTHORS.rst to include contributors graph by @0scvr in https://github.com/pyvista/pyvista/pull/5126
* Add explanation for adding test code for deprecated functions by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5120
* Add `Prop3D` class to the API document by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5085
* Add Preview the Documentation section in CONTRIBUTING by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5091
* Add API doc of Features by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5133
* Add GitHub Codespaces badge in README.rst and CONTRIBUTING.rst by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5123
* Deprecate plotter mesh update helpers by @banesullivan in https://github.com/pyvista/pyvista/pull/5158
* Fix trimesh intersphinx by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5187
* Remove `cell_arrays` in documentation by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5199
* Fix a small typo by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5225
* Deprecate/fix `PolyData.n_faces` by @darikg in https://github.com/pyvista/pyvista/pull/5229
* Fix docstring format error by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5243
* Move `input_validation` private package by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5247
* Fix docstring of `DataSet.cell_neighbors_levels` by @dionhaefner in https://github.com/pyvista/pyvista/pull/5293
### Maintenance
* Fix numpydoc ignore comment by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4840
* Download vtk-osmesa's wheel with aria2c by @eggplants in https://github.com/pyvista/pyvista/pull/4864
* Change `.pre-commit-config.yaml` interval weekly to monthly by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4874
* Improve test coverage of `Plotter.add_floor` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4866
* Fix typos using `typos` by @eggplants in https://github.com/pyvista/pyvista/pull/4865
* Remove unused fixtures in pytest by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4885
* Fix test functions names in `test_download_files.py` by @igmochang in https://github.com/pyvista/pyvista/pull/4925
* Rename GitHub Actions path pyansys to ansys by @MihirKohli in https://github.com/pyvista/pyvista/pull/4935
* Deprecate `add_pick_obeserver` and use `add_pick_observer` by @darikg in https://github.com/pyvista/pyvista/pull/4953
* Add `auto-approve.yml` to allow review request by bot by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4941
* Fix `test downloads` option errors by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4924
* Add `auto-approve-dependabot.yml` to auto approve Dependabot PR by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4940
* Update geovista integration test by @bjlittle in https://github.com/pyvista/pyvista/pull/4957
* Add `check-docstring-first` and `end-of-file-fixer` in `pre-commit-hooks` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4959
* Fix GeoVista's CI error due to image regerssion test by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4970
* Use `itertools.product` to avoid nested loops in `renderers.py` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4969
* Move translate test to `test_geometric_sources.py` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4978
* Add preview documentation in pull request by @tkoyama010 in https://github.com/pyvista/pyvista/pull/4934
* Trigger Pull Request Labeler after approved review by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5028
* Delete `.github/workflows/greetings.yml` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5026
* Allow deploying preview of the forked repository using  label by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5024
* Revert "Allow deploying preview of the forked repository using  label" by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5030
* Add define alias `pv` commit to `.git-blame-ignore-revs` file by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5037
* Revert "Trigger Pull Request Labeler after approved review" by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5038
* Fix ParametricRandomHills parameters names by PEP8 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5039
* Use `itertools.product` to avoid nested loops in `arrays.py` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5053
* Fix testing warnings in tests/core by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5049
* Revert "Add define alias `pv` commit to `.git-blame-ignore-revs` file" by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5070
* Add `_coerce_transformlike_arg` function by @user27182 in https://github.com/pyvista/pyvista/pull/5052
* Remove deprecations in v0.37.0 to remove at v0.43.0 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5080
* Deprecate `t_coords` and use `texture_coordinates` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5079
* Fix PyVistaDeprecationWarning not imported error by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5084
* Split Preview Documentation in GitHub Actions by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5083
* Defer to geovista default warning escalation to error with pytest by @bjlittle in https://github.com/pyvista/pyvista/pull/5081
* Revert "Split Preview Documentation in GitHub Actions" by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5093
* Remove deprecations in v0.32.0 to remove at v0.43.0 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5086
* Remove deprecations in v0.38.0 to remove at v0.43.0 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5088
* Remove deprecations in v0.40.0 to remove at v0.43.0 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5090
* Remove deprecations in v0.39.0 to remove at v0.43.0 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5089
* Remove deprecations in v0.35.0 to remove at v0.43.0 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5087
* Pin Vale version to version 2.29.5 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5109
* Add `Plotter.add_point_labels` `shape` argument test by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5110
* Fix `labelMapper` to `labe_mapper` following PEP8 rule by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5101
* Fix `.pre-commit-config.yaml` setting error by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5124
* Use -a for xvfb-run in Plotting Testing (uses GL) by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5134
* Revert "Add Preview the Documentation section in CONTRIBUTING" by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5139
* Fix VTK 9.3 failures by @banesullivan in https://github.com/pyvista/pyvista/pull/4688
* Remove unnecessary if statement to use `wrap` by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5143
* Update example deprecation code in `CONTRIBUTING.rst` by @user27182 in https://github.com/pyvista/pyvista/pull/5150
* Update .pre-commit-config.yaml to exclude Sphinx objects.inv files by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5167
* Clean up unused arguments by @banesullivan in https://github.com/pyvista/pyvista/pull/5147
* Enchance BUILDING_GALLERY flag code by @ASafarzadeh in https://github.com/pyvista/pyvista/pull/5189
* Support python 3.12 and vtk 9.3.0 in CI by @MatthewFlamm in https://github.com/pyvista/pyvista/pull/5196
* Use python 3.12 and vtk 9.3.0 in Build Documentation by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5202
* ` PyQt6-Qt6!=6.6.1` to fix PyVistaQt and MNE-Python integration tests by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5263
* Add command that preview forked pull request doc by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5265
* Update preview_pull_request.yml to enable-commit-comment false by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5274
* ` PyQt6!=6.6.1` to fix PyVistaQt and MNE-Python integration tests by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5279
* Add support for `trame_jupyter_extension` (`trame` over `jupyter` comm) by @alesgenova in https://github.com/pyvista/pyvista/pull/5240
* Update label.yml and labeler.yml to fix breaking changes in V5 by @tkoyama010 in https://github.com/pyvista/pyvista/pull/5287

## New Contributors
* @planmi1287 made their first contribution in https://github.com/pyvista/pyvista/pull/4856
* @FelipeCybis made their first contribution in https://github.com/pyvista/pyvista/pull/4855
* @eggplants made their first contribution in https://github.com/pyvista/pyvista/pull/4863
* @yr1404 made their first contribution in https://github.com/pyvista/pyvista/pull/4926
* @igmochang made their first contribution in https://github.com/pyvista/pyvista/pull/4925
* @MihirKohli made their first contribution in https://github.com/pyvista/pyvista/pull/4935
* @AdityaRanjanJha made their first contribution in https://github.com/pyvista/pyvista/pull/4946
* @ChristosT made their first contribution in https://github.com/pyvista/pyvista/pull/4958
* @Arnav-2004 made their first contribution in https://github.com/pyvista/pyvista/pull/4993
* @denizkirbiyik made their first contribution in https://github.com/pyvista/pyvista/pull/5003
* @abhiramtilakiiit made their first contribution in https://github.com/pyvista/pyvista/pull/5036
* @jourdain made their first contribution in https://github.com/pyvista/pyvista/pull/5045
* @jorgensd made their first contribution in https://github.com/pyvista/pyvista/pull/5072
* @roceb made their first contribution in https://github.com/pyvista/pyvista/pull/5078
* @0scvr made their first contribution in https://github.com/pyvista/pyvista/pull/5126
* @kmarchais made their first contribution in https://github.com/pyvista/pyvista/pull/5181
* @ASafarzadeh made their first contribution in https://github.com/pyvista/pyvista/pull/5189
* @bsburnham made their first contribution in https://github.com/pyvista/pyvista/pull/5095
* @alesgenova made their first contribution in https://github.com/pyvista/pyvista/pull/5240
* @dionhaefner made their first contribution in https://github.com/pyvista/pyvista/pull/5293

**Full Changelog**: https://github.com/pyvista/pyvista/compare/v0.42.0...v0.43.0

<hr /><em>This discussion was created from the release <a href='https://github.com/pyvista/pyvista/releases/tag/v0.43.0'>v0.43.0</a>.</em>

**Comment by akaszynski** [comment]
Excellent work @tkoyama010!

**Reply by MatthewFlamm** [reply]
Also an impressive number of new contributors in this release.  Excellent work by all contributors!

**Reply by tkoyama010** [reply]
There is also good news about PyVista's paper. As confirmed by [Dimensions](https://badge.dimensions.ai/), PyVista's paper has achieved excellent results in terms of citation performance.

Dimensions – PyVista: 3D plotting and mesh analysis through a streamlined interface for the Visualization Toolkit (VTK) https://badge.dimensions.ai/details/id/pub.1114813963 

> This publication in The Journal of Open Source Software has been cited 217 times. 76% of its citations have been received in the past two years, which is higher than you might expect, suggesting that it is currently receiving a lot of interest.
>
> Compared to other publications in the same field, this publication is extremely highly cited and has received approximately 81 times more citations than average.
